src/index.ts

```diff
@@ -0,0 +1,323 @@
+
+          // **************************************************************************
+          // 5. Make API request
+          // **************************************************************************
+          console.log("[DEBUG]\tmaking API request...");
+          const configOis = ois.find((o) => o.title === oisTitle)!;
+          const configEndpoint = configOis.endpoints.find(
+            (e) => e.name === endpointName
+          )!;
+          const templateParameters = safeDecode(template.parameters);
+          const reservedParameters = getReservedParameters(
+            configEndpoint,
+            templateParameters || {}
+          );
+          if (!reservedParameters._type) {
+            console.log("[ERROR]\treserved parameter 'type' is missing");
+            return;
+          }
+          const sanitizedParameters: adapter.Parameters = removeKeys(
+            templateParameters || {},
+            RESERVED_PARAMETERS
+          );
+          const adapterApiCredentials = apiCredentials
+            .filter((c) => c.oisTitle === oisTitle)
+            .map((c) => removeKey(c, "oisTitle") as adapter.ApiCredentials);
+
+          const options: adapter.BuildRequestOptions = {
+            ois: configOis,
+            endpointName,
+            parameters: sanitizedParameters,
+            metadataParameters: {}, // TODO: https://github.com/api3dao/airnode/pull/697
+            apiCredentials: adapterApiCredentials,
+          };
+
+          const apiResponse = await adapter.buildAndExecuteRequest(options);
+          if (!apiResponse || !apiResponse.data) {
+            console.log("[ERROR]\tfailed to fetch data from API");
+            return;
+          }
+          console.log("[INFO]\tAPI server response data:", apiResponse.data);
+          // TODO: should we really return here or 0 could be a valid response?
+          if (apiResponse.data === 0) {
+            console.log("[ERROR]\tAPI responded with value of 0");
+            return;
+          }
+
+          let apiValue: ethers.BigNumber;
+          try {
+            const response = adapter.extractAndEncodeResponse(
+              apiResponse.data,
+              reservedParameters as adapter.ReservedParameters
+            );
+            apiValue = ethers.BigNumber.from(
+              adapter.bigNumberToString(response.value as any)
+            );
+
+            console.log("[INFO]\tAPI server value:", apiValue.toNumber());
+          } catch (e) {
+            console.log("[ERROR]\tfailed to extract data from API response");
+            return;
+          }
+
+          // **************************************************************************
+          // 6. Read beacon
+          // **************************************************************************
+          console.log("[DEBUG]\treading beacon value from server...");
+          // address(0) is considered whitelisted
+          const voidSigner = new ethers.VoidSigner(
```

Comment:
this was pretty cool implementation, didn't know about this

Reply:
Yes indeed. All credit goes to Burak for finding this: https://api3workspace.slack.com/archives/C027Y2FAV0S/p1637666522074300